shell/mixins/resource-manager.js

```diff
@@ -0,0 +1,69 @@
+import { mapGetters } from 'vuex';
+import { allHashSettled } from '@shell/utils/promise';
+
+export default {
+  computed: { ...mapGetters(['currentCluster']) },
+  data() {
+    return { isLoadingSecondaryResources: false };
+  },
+  methods:  {
+    /**
+    *
+    * resourceData object structure
+    *
+    * @param {String} resourceData.namespace - Namespace identifier
+    * @param {Object} resourceData.data - Object containing info about the data needed to be fetched and how it should be parsed. Note: The KEY NEEDS to be the resource TYPE!
+    * @param {Array} resourceData.data[TYPE].applyTo - The array of operations needed to be performed for the specific data TYPE
+    * @param {String} resourceData.data[TYPE].plural - String to override the default 's' plural used in resource requests, if needed
+    * @param {String} resourceData.data[TYPE].applyTo[x].var - The 'this' property name that should be populated with the data fetched
+    * @param {Function} resourceData.data[TYPE].applyTo[x].parsingFunc - Optional parsing function if the fetched data needs to be parsed
+    */
+    async $resourceManagerFetchSecondaryResources(resourceData) {
+      const requests = {};
+      const clusterId = this.currentCluster.id;
+      const namespace = resourceData.namespace;
+
+      // Only fetch types if the user can see them
+      Object.keys(resourceData.data).forEach((type) => {
+        const schema = this.$store.getters['cluster/schemaFor'](type);
+
+        if (schema) {
+          const isNamespaced = schema?.attributes?.namespaced || false;
+          let url = `/k8s/clusters/${ clusterId }/api/v1/namespaces/${ namespace }`;
```

Comment:
It's better to avoid `lets` if possible
```
const url = isNamespaced ? `/k8s/clusters/${ clusterId }/api/v1/namespaces/${ namespace }` : `/k8s/clusters/${ clusterId }/v1`
```

However... i hit an issue where the resource was namespaced and `namespace` was null. So maybe best to validate and warn if so.

Reply:
The URL should come from the schema's `links.collection` link. If the schema is namespaced that can be injected before the resource type (the url can be `split('/')` and the `namespaces/${namespace}` value inserted before the last entry)

That would remove the need for any `plural` / `requestPlural` propertries

Reply:
The `schema.links.collection`, including domain, is a better source of truth for fetching collections than hard coded strings. The same schema `collection` value  is used in `urlFor` when we fetch resources normally.

I've tested below and it works fine. This approach also doesn't require the currentlCluster, which was injected into the hard coded string. If we were to use that in a cluster agnostic product like `cluster manager` it would error (where the cluster would be blank or a different one not related to the request we would be making).

```
          let url = schema.links.collection;

          if (schema?.attributes?.namespaced && namespace) {
            const parts = url.split('/');

            parts.splice(parts.length - 2, 0, `api`);
            parts.splice(parts.length - 1, 0, `namespaces/${ namespace }`);
            url = parts.join('/');
          }
```

Reply:
Thanks for the help @richard-cox 🙏